packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -184,6 +184,42 @@ export class ProjectActions {
     return this.api.launchProject(browser, spec, options)
   }
 
+  async launchProjectWithoutLaunchpad (options?: LaunchOpts) {
+    if (!this.ctx.activeProject) {
+      throw Error('Cannot launch project without an active project')
+    }
+
+    const preferences = await this.api.getProjectPreferencesFromCache()
+    const { browserId, testingType } = preferences[this.ctx.activeProject.title] ?? {}
+
+    if (!browserId || !testingType) {
+      throw Error('Cannont launch project without stored browserId or testingType')
+    }
+
+    const spec = this.makeSpec(testingType)
```

Comment:
I guess this is hard requirement for now, but eventually we should not need to do this, I think... - maybe we can add a TODO to remove this when we refactor `launchProject` to not require a spec on launch.

Also, since `makeSpec` isn't used anywhere else, should we just do something like:

```js
const spec = {
  name: '...'
  // ...
}
```

Reply:
I took this from the launchProject and forgot to add it back there. I can add this function there as well so when we refactor it to not require a spec we can clean it up in both spots.